rotkehlchen/externalapis/cowswap.py

```diff
@@ -93,8 +93,13 @@ def get_order_data(self, order_uid: str) -> tuple[int, str]:
                 order_type = data['class']
             else:
                 raw_fee_amount = int(data.get('executedSurplusFee', data.get('executedFee', 0)))
-                metadata = json.loads(data['fullAppData'])['metadata']
-                order_type = metadata['orderClass']['orderClass'] if 'orderClass' in metadata else 'limit'  # noqa: E501
+
+                try:
+                    # Cowswap API sometimes returns nullish `fullAppData` so this may fail
+                    metadata = json.loads(data['fullAppData'])['metadata']
```

Comment:
It's better to turn this into a null check. Something like

```python
if (appdata := data.get('fullAppData')) is not None:
    metadata = json.loads(data['fullAppData'])['metadata']
    order_type = metadata['orderClass']['orderClass'] if 'orderClass' in metadata else 'limit'  # noqa: E501
else:
    order_type = 'limit'
```

Also is this new? How come it happens now? We have never seen this in production yet though I for example use cowswap daily